### PR TITLE
SmartThings integration option: events

### PR DESCRIPTION
**Pull request by andrewsayre**
**Description:**

Documents the integration option "Raise events for push updates" and the new event type `smartthings.update` in the SmartThings.

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#26499

## Checklist:

- [X] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [X] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Comment by frenck** [comment]
@andrewsayre Please don't add the `awaits-parent` label. This is used to indicate it had a review and cannot be merged yet. Since your PR hasn't had a review yet, this label is incorrect and should not be applied.

File: source/_components/smartthings.markdown [hunk 1]
```diff
@@ -106,6 +106,12 @@ Advanced: If you have multiple locations in SmartThings, each can be integrated
 
 See the [troubleshooting](#troubleshooting) if you are having issues setting up the integration.
 
+## Integration Options
+
+The SmartThings integration supports the following options (configured through the integration options screen):
+
+1. `Raise events for push updates`: This is an advanced feature which provides access to the raw data received from SmartThings during a push update. When enabled, the integraiton will raise events through the event bus for all push updates--not just button presses. The default is to raise events only for buttons.
```
Comment (frenck) [inline source/_components/smartthings.markdown]:
```suggestion
1. `Raise events for push updates`: This is an advanced feature which provides access to the raw data received from SmartThings during a push update. When enabled, the integration will raise events through the event bus for all push updates--not just button presses. The default is to raise events only for buttons.
```

File: source/_components/smartthings.markdown [hunk 2]
```diff
@@ -134,6 +140,36 @@ The integration will trigger an event when a device with the [button](https://sm
...
+| Attribute                 | Description
+|---------------------------|------------------------------------------------------------------|
+`location_id`               | The unique id of the location the device is part of. This can be found in the config entry registry or in the [SmartThings Groovy IDE](https://developers.smartthings.com/).
+`device_id`                 | The unique id of the device in SmartThings. This can be located in the HASS device registry or in the [SmartThings Groovy IDE](https://developers.smartthings.com/).
+`component_id`              | Describes which integration of the device triggered the event. `main` represents the parent device. For devices with child-devices, this attribute identifies the child that raised the event.
+`name`                      | The name given to the device in SmartThings.
+`capability`                | The SmartThings capability that was updated.
+`attribute`                 | The attribute of the capability that was updated.
+`value`                     | The new value of the attribute.
+`data`                      | Data assocaited with the event update.
```
Comment (frenck) [inline source/_components/smartthings.markdown]:
```suggestion
`data`                      | Data associated with the event update.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Reviewed, 2 small typos.

File: source/_components/smartthings.markdown (hunk 1 above)
Comment (andrewsayre) [reply source/_components/smartthings.markdown]: Fixed!

File: source/_components/smartthings.markdown (hunk 2 above)
Comment (andrewsayre) [reply source/_components/smartthings.markdown]: Fixed!

**Comment by andrewsayre** [comment]
@frenck sorry about that...  I've been using that label wrong this whole time.  :joy:

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Pull request closed without merging**